ProjectSource/app-library/src/test/java/com/kayako/sdk/android/k5/core/KayakoLogHelperTest.java

```diff
@@ -0,0 +1,77 @@
+package com.kayako.sdk.android.k5.core;
+
+import static org.hamcrest.CoreMatchers.is;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ErrorCollector;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.Mock;
+import static org.mockito.Mockito.verify;
+import org.mockito.junit.MockitoJUnitRunner;
+import java.util.List;
+
+@RunWith(MockitoJUnitRunner.class)
+public class KayakoLogHelperTest {
+
+    @Mock
+    private KayakoLogHelper.PrintLogListener printLogListener;
+
+    @Mock
+    private Throwable throwable;
+
+    @Captor
+    private ArgumentCaptor<String> stringArgumentCaptor;
+
+    @Captor
+    private ArgumentCaptor<Throwable> throwableArgumentCaptor;
+
+    @Rule
+    public final ErrorCollector errorCollector = new ErrorCollector();
+
+    @Test
+    public void test() {
+        //Arrange
+        final String debugTag = "debugTag";
+        final String debugMessage = "debugMessage";
+        final String verboseTag = "verboseTag";
+        final String verboseMessage = "verboseMessage";
+        final String errorTag = "errorTag";
+        final String errorMessage = "errorMessage";
+        final String stackTraceTag = "stackTraceTag";
+        final String exceptionTag = "exceptionTag";
+
+        //Act
+        KayakoLogHelper.addLogListener(printLogListener);
+        KayakoLogHelper.d(debugTag, debugMessage);
+        KayakoLogHelper.v(verboseTag, verboseMessage);
```

Comment:
❗️ This code could improve SRP (from SOLID) and the FIRST testing principles compliance. FIRST states a test should be "I"- isolated. Consider the suggested modifications in order to avoid this violation. 

**[More details](https://confluence.devfactory.com/x/NAQGF)**
          
## The solution in this case

Creating separate test methods for each log method:

```Java
     @Test 
     public void givenKayakoLogHelperWhenDThenParametersArePassed() { 

     @Test 
     public void givenKayakoLogHelperWhenVThenParametersArePassed() { 

     @Test 
     public void givenKayakoLogHelperWhenEThenParametersArePassed() { 

     @Test 
     public void givenKayakoLogHelperWhenPrintStackTraceThenParametersArePassed() { 
     ...
```
 

<a id="9da86820-4949-11e8-a394-d5f05418fe24" href="none" codereview-uuid-reserved-tag />